packages/uxpin-merge-cli/src/common/services/UXPin/updateRepositoryPointerToBranchOrTag.ts

```diff
@@ -24,9 +25,34 @@ export async function updateRepositoryPointerToBranch(
 
   if (!opts) { throw new Error('Missing/invalid options'); }
   if (!opts.apiDomain) { throw new Error('Missing API domain for repository pointer update'); }
-  if (!opts.branch) { throw new Error('Missing branch name for repository pointer update'); }
   if (!opts.commitHash) { throw new Error('Missing commit hash for repository pointer update'); }
   if (!opts.authToken) { throw new Error('Missing auth token for repository pointer update'); }
+  if (!opts.branch) { throw new Error('Missing branch name for repository pointer update'); }
+
+  // If an optional tag was provided then we need to make a duplicate repository
+  // pointer with type 'tag' that points to the same commit hash
+  if (opts.tag) {
+    try {
+      await createTag(opts)
+        .then(() => undefined);
+    } catch (error) {
+      // Could throw error if the tag already exists
+      throw new Error(error.message);
+    }
+  }
+
+  return updateRepositoryPointerToBranch(opts)
+    .then(() => undefined);
```

Comment:
I think `createTag` should be independent from `updateRepositoryPointer`? 🤔  because it shouldn't set a pointer to the tag from this action.
Can we create a separate file for this and call it from `uploadLibrary`?

Reply:
Yep - agreed :smile: